File: sample-projects/typescript/src/de/sots/cellarsandcentaurs/domain/model/Dice.ts
export class DiceRoll {
    constructor(
        readonly sides: number,
        readonly value: number,
    ) {}
}

export class Dice {
    constructor(private readonly sides: number) {}

    roll(): DiceRoll {
        const value = Math.floor(Math.random() * this.sides) + 1;
        return new DiceRoll(this.sides, value);
    }
}

export function rollD20(): number {
    const d20Roll = new Dice(20).roll();
    return d20Roll.value;
}

File: sample-projects/python/de/sots/cellarsandcentaurs/domain/model/dice.py
import random
from dataclasses import dataclass


@dataclass(frozen=True)
class Dice:
    sides: int

    def roll(self) -> "DiceRoll":
        return DiceRoll(self, random.randint(1, self.sides))


@dataclass(frozen=True)
class DiceRoll:
    dice: Dice
    value: int


def roll_d20() -> DiceRoll:
    d20Roll = Dice(20).roll()
    return d20Roll
